context/http.go

```diff
@@ -17,11 +17,34 @@ var (
 	ErrNoRequestContext = errors.New("no http request in context")
 )
 
+// http.Request.RemoteAddr is not the originating IP if the request
+// came through a reverse proxy.
+// X-Forwarded-For and X-Real-IP provide this information with decreased
+// support, so check in that order.  There may be multiple 'X-Forwarded-For'
+// headers, but go maps don't retain order so just pick one.
+func RemoteAddr(r *http.Request) string {
+	remoteAddr := r.RemoteAddr
+
+	if prior, ok := r.Header["X-Forwarded-For"]; ok {
+		proxies := strings.Split(prior[0], ",")
+		if len(proxies) > 0 {
+			remoteAddr = strings.Trim(proxies[0], " ")
+		}
+	} else if realip, ok := r.Header["X-Real-Ip"]; ok {
+		if len(realip) > 0 {
+			remoteAddr = realip[0]
+		}
+	}
```

Comment:
remoteAddr is initialized to r.RemoteAddr.  If neither header exists, the value remains the same.